lib/mocha/configuration.rb

```diff
@@ -303,6 +306,17 @@ def reinstate_undocumented_behaviour_from_v1_9?
       @options[:reinstate_undocumented_behaviour_from_v1_9]
     end
 
+    # @private
+    def strict_keyword_argument_matching=(value)
+      raise "Strict keyword argument matching requires Ruby 2.7 and above." unless Mocha::RUBY_V27_PLUS
```

Comment:
@floehopper how would you suggest handling a config that only works in certain versions of Ruby? Should we raise during the configuration (my preference, as it makes the issue obvious earlier), or log some warning instead?

Reply:
I don't think we've run into this issue before. I'm definitely in favour of failing fast.

I was wondering whether we should introduce a new exception class (e.g. `Mocha::ConfigurationError`), but there is already precedence for just using `StandardError` like you've done in e.g. `lib/mocha/integration.rb`, so I think that's fine. And the stack trace ought to make it clear where the problem lies.

One option I've just thought of is to only define the relevant configuration-option-related method(s) if a compatible version of Ruby is available...? However, it might be less obvious if the user sees a `NoMethodError`, so perhaps your approach is better.